### PR TITLE
feat: add method GetImplicitResourcesForUser

### PR DESCRIPTION
**Pull request by closetool**
Signed-off-by: closetool <4closetool3@gmail.com>

Fix: #796

### CODE REVIEW

**Comment by closetool** [comment]
@hsluoyz This method just loop rmMap ,that might not right https://github.com/casbin/casbin/blob/5bdbb063721697233f9128e459a68399888f7bcb/rbac_api.go#L188

**Comment by hsluoyz** [comment]
@closetool we don't need new function. We just need to support g2 in GetImplicitPermissionsForUser().

**Comment by closetool** [comment]
@hsluoyz Supporting `g2` could not satisfy customer's requirement 
https://github.com/casbin/jcasbin/issues/174#issuecomment-822528302
maybe I'll make a pr later to support `g2` in `GetPermissionsForUser`

**Comment by hsluoyz** [comment]
@closetool rename `GetImplicitPoliciesForUser()` to `GetImplicitResourcesForUser()`. Also plz add some tests.

**Comment by closetool** [comment]
Fixed @hsluoyz

**Comment by hsluoyz** [comment]
@closetool commit and PR name

File: rbac_api.go [hunk 1]
```diff
@@ -203,6 +203,36 @@ func (e *Enforcer) GetImplicitRolesForUser(name string, domain ...string) ([]str
 	return res, nil
 }
 
+func (e *Enforcer) GetImplicitUsersForRole(name string, domain ...string) ([]string, error) {
```
Comment (RobotHuang) [inline rbac_api.go:206]: plz add some tests for it

**Comment by hsluoyz** [comment]
@lanmarti plz review. Although this is Go code, but if it looks OK, we will port it to Java.

**Review by RobotHuang — APPROVED** [review]

File: rbac_api.go [hunk 2]
```diff
@@ -203,6 +203,37 @@ func (e *Enforcer) GetImplicitRolesForUser(name string, domain ...string) ([]str
...
+			}
+			for _, r := range roles {
+				if _, ok := roleSet[r]; !ok {
+					res = append(res, r)
+					q = append(q, r)
+					roleSet[r] = true
+				}
+			}
+		}
+	}
```
Comment (lanmarti) [inline rbac_api.go:232]: I'm unfamiliar with go, so I'm probably reading it wrong, but does this mean that the following would be true?
```
g, alice, subgroup1
g2, subgroup1, topgroupA
```
results in GetImplicitUsersForRole(topgroupA) = [subgroup1, alice] ? And if so, would this not invalidate group isolation?

Comment (closetool) [reply rbac_api.go:232]: @lanmarti @hsluoyz That a egacy issue.

**Review by lanmarti — APPROVED** [review]

**Comment by hsluoyz** [comment]
@closetool plz add the new API to docs.

**Pull request merged**